src/dependencies/formatter/direct_deps.rs

```diff
@@ -1,135 +1,107 @@
+use crate::config::OutputFormat;
 use crate::dependencies::formatter::{
     format_version, get_package_metadata_msrv, parse_manifest_workaround,
 };
 use crate::dependencies::DependencyGraph;
-use crate::reporter::Output;
-use std::marker::PhantomData;
-
-pub(crate) struct DirectDependenciesFormatter<T: Output> {
```

Comment:
maybe it's helpful if i explain my reasoning here?

i removed this class because i think it's needlessly stateful, and quite convoluted.

- two output styles are implemented using closures, but since they don't share any code i've just split them into separate functions
- the 'stateful' `*Formatter` classes don't really provide any value here. At the end of the day, their only purpose is to be called once and output a `String`.  This is better handled by a plain function. If the output functions were re-entrant in some way it might make sense to use a stateful formatter. but that's not the case here.
- i've also managed to remove a few allocations by chaining iterators. Not that i think that's ever going to be a performance bottleneck in this crate....